packages/client/src/forms/configurable/index.ts

```diff
@@ -0,0 +1,12514 @@
+            }
+          }
+        },
+        {
+          name: 'nid',
+          type: TEXT,
+          label: userMessages.NID,
+          required: false,
+          initialValue: '',
+          validate: [
+            {
+              operation: 'validIDNumber',
+              parameters: [NATIONAL_ID]
+            }
+          ],
+          mapping: {
+            mutation: {
+              operation: 'fieldToIdentifierWithTypeTransformer',
+              parameters: [NATIONAL_ID]
+            },
+            query: {
+              operation: 'identifierWithTypeToFieldTransformer',
+              parameters: [NATIONAL_ID]
+            }
+          }
+        },
+        {
+          name: 'accountDetails',
+          type: FIELD_GROUP_TITLE,
+          label: userMessages.accountDetails,
+          required: false,
+          initialValue: '',
+          validate: []
+        },
+        {
+          name: 'role',
+          type: SELECT_WITH_OPTIONS,
+          label: userMessages.labelRole,
+          required: true,
+          initialValue: '',
+          validate: [],
+          options: []
+        },
+        {
+          name: 'type',
+          type: SELECT_WITH_DYNAMIC_OPTIONS,
+          label: userMessages.type,
+          required: true,
+          initialValue: '',
+          validate: [],
+          dynamicOptions: {
+            dependency: 'role',
+            options: {}
+          }
+        },
+        {
+          name: 'device',
+          type: TEXT,
+          label: userMessages.userDevice,
+          required: false,
+          initialValue: '',
+          validate: []
+        }
+      ]
+    },
+    {
+      id: 'signature-attachment',
+      title: userMessages.userSignatureAttachmentTitle,
+      conditionals: [
+        {
+          action: 'hide',
+          expression: 'values.role!=="LOCAL_REGISTRAR"'
+        }
+      ],
+      fields: [
+        {
+          name: 'attachmentTitle',
+          type: FIELD_GROUP_TITLE,
+          hidden: true,
+          label: userMessages.userAttachmentSection,
+          required: false,
+          initialValue: '',
+          validate: []
+        },
+        {
+          name: 'signature',
+          type: SIMPLE_DOCUMENT_UPLOADER,
+          label: userMessages.userSignatureAttachment,
+          description: userMessages.userSignatureAttachmentDesc,
+          allowedDocType: ['image/png'],
+          initialValue: '',
+          required: true,
+          validate: []
+        }
+      ]
+    }
+  ]
+}
+
+const getPreviewGroups = () => {
+  return userSectionFormType.groups.map((group) => {
+    return {
+      id: `preview-${group.id}`,
+      fields: group.fields
+    }
+  }) as IFormSectionGroup[]
+}
+
+export const userSectionPreviewType: IFormSection = {
+  id: UserSection.Preview,
+  viewType: 'preview',
+  name: userMessages.userFormReviewTitle,
+  title: userMessages.userFormTitle,
+  groups: getPreviewGroups()
+}
+
+export const defaultFormsConfig = {
```

Comment:
I want to add the type IForms here.  But if I do, I have issues with mapping and validate types that I cant resolve